rdt/transformers/numerical.py

```diff
@@ -151,14 +150,11 @@ def reverse_transform(self, data):
         if self.nan is not None:
             data = self.null_transformer.reverse_transform(data)
 
-        if self._rounding_digits is not None:
-            data = data.round(self._rounding_digits)
+        is_integer = np.dtype(self._dtype).kind == 'i'
+        if self._rounding_digits is not None or is_integer:
+            data = data.round(self._rounding_digits or 0)
```

Comment:
if `_rounding_digits` is not None, why do we need to do `or 0`?

Reply:
Because if `is_integer` is `True` we enter this branch even if `self._rounding_digits` is `None`. In that case, we want to round to `0` rather than `None`.